test/functional/api/vim_spec.lua

```diff
@@ -657,6 +657,114 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_list_chans and nvim_get_chan_info', function()
+    before_each(function()
+      command('autocmd ChanOpen * let g:opened_event = copy(v:event)')
+      command('autocmd ChanInfo * let g:info_event = copy(v:event)')
+    end)
+    local testinfo = {
+      stream = 'stdio',
+      id = 1,
+      mode = 'rpc',
+      client = {},
+    }
+    local stderr = {
+      stream = 'stderr',
+      id = 2,
+      mode = 'bytes',
+    }
+
+    it('returns {} for invalid channel', function()
```

Comment:
I like this for "get". But shouldn't "set" throw an error for invalid channel?

Reply:
You can only set for your own channel. If the channel is invalid, this is caught before the API method is even entered.

Reply:
Ok. Is there no other validation that should be done by `nvim_set_client_info` ?

Reply:
We can't check against invalid keys, as new keys might be added. We could add a check for `major` version, or perhaps set it to `0` per default?